src/dwarf.cpp

```diff
@@ -644,6 +736,56 @@ pool_string dwarf::implementation::read_debug_str(std::size_t offset) {
                                                 [&] { return empool(_s.read_c_string_view()); });
 }
 
+/**************************************************************************************************/
+// SPECREF: DWARF5 page 26 (8) line 28 -- v4 -> v5 changes
+pool_string dwarf::implementation::read_debug_str_offs(std::size_t entry) {
+    if (const auto found = _debug_str_offs_cache.find(entry); found != _debug_str_offs_cache.end()) {
+        return found->second;
+    }
+
+    // SPECREF: DWARF5 page 259 (241) line 6 --
+    // Apparently `DW_AT_str_offsets_base` points to the first entry in this table, but I am not
+    // sure where that attribute lives. So we'll take the time to derive that offset every time.
+    // If that becomes too expensive we can revisit hunting down `DW_AT_str_offsets_base` and
+    // caching it.
+
+    // This section contains a header, then a series of offsets stored as 4- or 8-byte
+    // values, then a series of strings. So we have to jump twice: first to get
+    // the offset, then to get the string. The 0th string immediately follows the last
+    // entry offset.
+
+    const std::size_t entry_offset = temp_seek(_s, _debug_str_offsets._offset, [&] {
```

Comment:
do you need to capture anything?

Reply:
I'm not quite sure what you mean with the question? I'm using the lambda to convert the string entry that comes in to an offset with a specific DWARF5 data section.

Reply:
Oh capturing within the lambda? The `[&]` gives the compiler the freedom to capture whatever it needs by reference. Since this lambda's lifetime is limited to the `temp_seek` call, everything its referencing is alive when used within the lambda.